app/assets/javascript/app/components/fields/MonthYearField.jsx

```diff
@@ -54,6 +118,8 @@ class MonthYearField extends React.Component {
   }
 
   render() {
+    this.setTimeConstraints();
```

Comment:
Any function that sets or evaluates state should not be in the render function because it could be called needlessly on re-renderings or, more dangerously, cause an infinite loop where setState causes another render.

The options should be limited and the existing values should be "snapped" in componentWillReceiveProps when the predate or postdate change.

Reply:
I couldn't find a lifecycle method other than `render` that gets called when the state changes. `setTimeConstraints` only updates the state conditionally but must be called when the state changes. This is due to the month-year pairs deciding which limits exist if one or both have already been modified. 

Is there a lifecycle method that will trigger when state changes other than render?